scipy/optimize/_slsqp_py.py

```diff
@@ -498,20 +498,16 @@ def cjac(x, *args):
         majiter_prev = int(majiter)
 
     # Obtain KKT multipliers from internals of SLSQP (see slsqp.slsqp_optmz.f)
-    im = 1
-    il = im + la
-    ix = il + (n1 * n) // 2 + 1
-    # Start of constraint KKT multipliers in w
+    im = 0  # fortran indexing starts at 1 but python starts at 0
+    il = im + m
+    ix = il + (n1*n)//2 + 1
     ir = ix + n
-    # Subtract 1 since fortran indexing starts at 1 but python starts at 0
-    ir -= 1
 
     _kkt_mult = w[ir:ir + m]
 
     # Indices for bound constraints (one per variable).
-    # Found by trial and error since documentation is inconsistent
-    iu = ir + n + n + la + n1 # index iu from slsqp.slsqp_optmz.f
-    ib = iu + la + n - 2
+    # Found with trial and error since documentation is unclear
+    ib = ir + m + 4*n + 2
```

Comment:
Apparently, this seemed right enough to commit them, but it turned out that they were wrong? If that's the case, are we much more confident now that these are correct (e.g. we won't find out later that they are wrong, too)?

Reply:
This was found by trial and error based on a few tests which happened to have convenient values of `n` (number of parameters) and `m` (number of constraints) that worked out to be this way.

Upon noticing this, I implemented `test_kkt_bounds` to see what happens when you combine different values of `n` and `m`. This test and others in the suite helped work out the new index, which I think is adequately tested with `test_kkt_bounds`. We could try to come up with another similar test to make sure?